plugins/package-curation-providers/clearly-defined/src/funTest/kotlin/ClearlyDefinedPackageCurationProviderFunTest.kt

```diff
@@ -91,21 +110,36 @@ class ClearlyDefinedPackageCurationProviderFunTest : WordSpec({
             // Use an id which is known to have non-empty results from an earlier test.
             val packages = createPackagesFromIds("Maven:org.slf4j:slf4j-log4j12:1.7.30")
 
-            val curations = provider.getCurationsFor(packages)
+            withRetry {
+                val curations = provider.getCurationsFor(packages)
 
-            curations should beEmpty()
+                curations should beEmpty()
+            }
         }
 
         "be retrieved for packages without a namespace" {
             val provider = ClearlyDefinedPackageCurationProvider()
             val packages = createPackagesFromIds("NPM::acorn:0.6.0")
 
-            val curations = provider.getCurationsFor(packages)
+            withRetry {
+                val curations = provider.getCurationsFor(packages)
 
-            curations shouldNot beEmpty()
+                curations shouldNot beEmpty()
+            }
         }
     }
 })
 
 private fun createPackagesFromIds(vararg ids: String) =
     ids.map { Package.EMPTY.copy(id = Identifier(it)) }
+
+@OptIn(ExperimentalTypeInference::class)
+@ExperimentalKotest
+private suspend fun <T> withRetry(@BuilderInference test: suspend () -> T): T =
+    eventually(
```

Comment:
Any reason you're not using Kotest's [built-in `retry`](https://kotest.io/docs/assertions/retry.html), which seems to be a better fit?

Reply:
I wasn't aware of it. Don't know why you believe retry is a better fit. I've updated it, the code is slightly longer now.

Reply:
> Don't know why you believe retry is a better fit.

Because `retry`'s docs say "Retry is similar to eventually, but rather than attempt a block of code for a period of time, it attempts a block of code a maximum number of times", which IMO is exactly what we want.

Reply:
> I've updated it, the code is slightly longer now.

Hmm, it actually looks shorter to me...

Reply:
Sure, you're right.